flink-python/src/main/java/org/apache/flink/table/runtime/operators/python/aggregate/arrow/batch/BatchArrowPythonOverWindowAggregateFunctionOperator.java

```diff
@@ -0,0 +1,329 @@
+		}
+	}
+
+	@Override
+	public FlinkFnApi.UserDefinedFunctions getUserDefinedFunctionsProto() {
+		FlinkFnApi.UserDefinedFunctions.Builder builder = FlinkFnApi.UserDefinedFunctions.newBuilder();
+		// add udaf proto
+		for (int i = 0; i < pandasAggFunctions.length; i++) {
+			FlinkFnApi.UserDefinedFunction.Builder functionBuilder =
+				getUserDefinedFunctionProto(pandasAggFunctions[i]).toBuilder();
+			functionBuilder.setWindowIndex(aggWindowIndex[i]);
+			builder.addUdfs(functionBuilder);
+		}
+		builder.setMetricEnabled(getPythonConfig().isMetricEnabled());
+		// add windows
+		for (int i = 0; i < lowerBoundary.length; i++) {
+			FlinkFnApi.Window.Builder windowBuilder = FlinkFnApi.Window.newBuilder();
+			if (isRangeWindows[i]) {
+				// range window
+				if (lowerBoundary[i] != Long.MIN_VALUE) {
+					if (upperBoundary[i] != Long.MAX_VALUE) {
+						// range sliding window
+						windowBuilder.setWindowType(FlinkFnApi.Window.WindowType.RANGE_SLIDING);
+					} else {
+						// range unbounded following window
+						windowBuilder.setWindowType(FlinkFnApi.Window.WindowType.RANGE_UNBOUNDED_FOLLOWING);
+					}
+				} else {
+					if (upperBoundary[i] != Long.MAX_VALUE) {
+						// range unbounded preceding window
+						windowBuilder.setWindowType(FlinkFnApi.Window.WindowType.RANGE_UNBOUNDED_PRECEDING);
+					} else {
+						// range unbounded window
+						windowBuilder.setWindowType(FlinkFnApi.Window.WindowType.RANGE_UNBOUNDED);
+					}
+				}
+			} else {
+				// row window
+				if (lowerBoundary[i] != Long.MIN_VALUE) {
+					windowBuilder.setLowerBoundary(lowerBoundary[i]);
+					if (upperBoundary[i] != Long.MAX_VALUE) {
+						// row sliding window
+						windowBuilder.setUpperBoundary(upperBoundary[i]);
+						windowBuilder.setWindowType(FlinkFnApi.Window.WindowType.ROW_SLIDING);
+					} else {
+						// row unbounded following window
+						windowBuilder.setWindowType(FlinkFnApi.Window.WindowType.ROW_UNBOUNDED_FOLLOWING);
+					}
+				} else {
+					if (upperBoundary[i] != Long.MAX_VALUE) {
+						// row unbounded preceding window
+						windowBuilder.setUpperBoundary(upperBoundary[i]);
+						windowBuilder.setWindowType(FlinkFnApi.Window.WindowType.ROW_UNBOUNDED_PRECEDING);
+					} else {
+						// row unbounded window
+						windowBuilder.setWindowType(FlinkFnApi.Window.WindowType.ROW_UNBOUNDED);
+					}
+				}
+			}
+			builder.addWindows(windowBuilder);
+		}
+		return builder.build();
+	}
+
+	@Override
+	public String getFunctionUrn() {
+		return PANDAS_BATCH_OVER_WINDOW_AGG_FUNCTION_URN;
+	}
+
+	@Override
+	public String getInputOutputCoderUrn() {
+		return SCHEMA_OVER_WINDOW_ARROW_CODER_URN;
+	}
+
+	private boolean isInCurrentOverWindow(RowData data, long time, boolean includeEqual) {
+		long dataTime = data.getTimestamp(inputTimeFieldIndex, 3).getMillisecond();
+		long diff = time - dataTime;
+		return diff > 0 && order || diff == 0 && includeEqual;
```

Comment:
```suggestion
		return (diff > 0 && order) || (diff == 0 && includeEqual);
```
It makes the code more readable.